lochness/keyring/__init__.py

```diff
@@ -24,6 +26,79 @@ def box_api_token(Lochness, key):
             Lochness['keyring'][key]['API_TOKEN'])
 
 
+def mediaflux_api_token(Lochness, key):
+    """
+    Get mediaflux login credentials from an encrypted keyring file
+
+    Parameters
+    ----------
+    Lochness : dict
+        It is the configuration obtained from a file provided via scripts/sync.py --config
+
+    key : str
+        It is the subsection within the keyring_file where login credentials are noted
+        e.g. mediaflux.sydney. The latter is the remote name of the study under
+        Mediaflux column in STUDY_metadata.csv file. Meanwhile, keyring_file is listed
+        in the above configuration file.
+
+        Content of STUDY_metadata.csv:
+
+        Active,Consent,Subject ID,Mediaflux,XNAT
+        1,2017-02-09,sub01234,mediaflux.bwh:01234,xnat:HCPEP-BWH:01234
+        1,2017-02-09,sub01235,mediaflux.bwh:01235,xnat:HCPEP-BWH:01235
+        1,2017-02-09,sub01236,mediaflux.bwh:01236,xnat:HCPEP-BWH:01236
+
+        Content of keyring_file:
+
+        {
+            "lochness": {
+                "SECRETS": {
+                    "BWH":""
+                }
+            },
+            "mediaflux.bwh": {
+                "HOST": "mediaflux.researchsoftware.unimelb.edu.au",
+                "PORT": "443",
+                "TRANSPORT": "https",
+                "TOKEN":
+                "DOMAIN": "",
+                "USER": "",
+                "PASSWORD": ""
+            }
+        }
+
+    Returns
+    -------
+    : tuple
+        Login credentials for the remote host specified in the subsection
+    """
+
+    import os
+
+    if key not in Lochness['keyring']:
+        raise KeyringError('\'{0}\' not in keyring'.format(key))
+
+    # allow having both TOKEN and PASSWORD in keyring, prioritize TOKEN
+    if 'TOKEN' in Lochness['keyring'][key]:
+        credentials = ['HOST', 'PORT', 'TRANSPORT', 'TOKEN']
+    else:
+        credentials = ['HOST', 'PORT', 'TRANSPORT', 'TOKEN', 'DOMAIN', 'USER', 'PASSWORD']
+
+    auths= Lochness['keyring'][key]
+    mflux_cfg= os.path.join(Lochness['phoenix_root'], 'mflux.cfg')
```

Comment:
Is there a specific reason why you are creating an extra file tha thas the credentials?

Reply:
Yes, there is a specific reason. Mediaflux command line client would only accept a file to read login credentials from. There is no way we can provide them through the command line.